### PR TITLE
[Yaml] ensure dump indentation to be greather than zero

### PR DESCRIPTION
**Pull request by xabbuh**
| Q | A |
| --- | --- |
| Branch | master |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | https://github.com/symfony/symfony/pull/17943#issuecomment-190881815, #17977 |
| License | MIT |
| Doc PR |  |

This completes #17978.

### CODE REVIEW

File: src/Symfony/Component/Yaml/Dumper.php [hunk 1]
```diff
@@ -30,6 +30,10 @@ class Dumper
      */
     public function __construct($indentation = 4)
     {
+        if ($indentation < 1) {
+            throw new \InvalidArgumentException('The indentation must be greater than zero.');
```
Comment (dosten) [inline src/Symfony/Component/Yaml/Dumper.php]: The identation can be a float number like 3.14? or should ensure that the indentation is a natural number?

Comment (xabbuh) [inline src/Symfony/Component/Yaml/Dumper.php]: The docblock states that the argument must be an integer and we don't do scalar type checks in Symfony.

**Comment by fabpot** [comment]
Done in the merge commit. Thanks.

**Comment by xabbuh** [comment]
Thanks @fabpot!

**Pull request closed without merging**